fix(rdb): new troubleshooting pages (#4315)
scaleway/docs-content: pages/managed-databases-for-postgresql-and-mysql/troubleshooting/database-instance-performance-issues.mdx

---
meta:
  title: Dealing with Database Instance performance issues
  description: Troubleshoot Database Instance performance issues in Managed Databases for MySQL and PostgreSQL.
content:
  h1: Dealing with Database Instance performance issues
  paragraph: Troubleshoot Database Instance performance issues in Managed Databases for MySQL and PostgreSQL.
tags: disk-full databases
dates:
  validation: 2025-01-29
  posted: 2025-01-29
categories:
  - managed-databases
  - postgresql-and-mysql
---

## Problem

My Database Instance is not performing as expected.

## Possible causes

- High data loads
- High incoming traffic from specific apps or websites

## Solution

You can carry out the following actions:

1. [Monitor usage metrics on Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit) to:

>     - Check if the usage is regular. Database Instance can be impacted by high data loads or unexpected high traffic from an app or website, for example. You can [upgrade your Instance](/managed-databases-for-postgresql-and-mysql/how-to/upgrade-a-database) if necessary.

Reviewer: ```suggestion
    - Check if the usage is regular. The Database Instance can be impacted by high data loads or unexpected high traffic from an app or website, for example. You can [upgrade your Instance](/managed-databases-for-postgresql-and-mysql/how-to/upgrade-a-database) if necessary.
```

>     - Check if the usage is regular. Database Instance can be impacted by high data loads or unexpected high traffic from an app or website, for example. You can [upgrade your Instance](/managed-databases-for-postgresql-and-mysql/how-to/upgrade-a-database) if necessary.

Reviewer 2: ```suggestion
    - Check if the usage is nominal. Database Instance can be impacted by high data loads or unexpected high traffic from an app or website, for example. You can [upgrade your Instance](/managed-databases-for-postgresql-and-mysql/how-to/upgrade-a-database) if necessary.
```

>     - Check if there are slow queries in the Database Instance log. Huge or sub-optimized queries can fill up all the memory and trigger OOM kills

Reviewer 2: ```suggestion
    - Check if there are slow queries in the Database Instance log. Huge or sub-optimized queries can slow down the instance significantly, or even fill up all the memory and trigger OOM kills.
```

> 2. Try connecting to the Database Instance in a [Private Network](/managed-databases-for-postgresql-and-mysql/how-to/connect-database-private-network). Public endpoint traffic for Managed Databases for MySQL and PostgreSQL goes through a load balancer, which might add significant latency.

Reviewer 2: I would remove the "might". It definitely, in 100% of the case add **a lot** of latency. We not proud of it :see_no_evil:
```suggestion
2. Try connecting to the Database Instance in a [Private Network](/managed-databases-for-postgresql-and-mysql/how-to/connect-database-private-network). Public endpoint traffic for Managed Databases for MySQL and PostgreSQL goes through a load balancer, which add significant latency. it also ensure a better security, which makes it the recommended way to connect to a managed database.
```

content:
  h1: Dealing with Database Instance performance issues
  paragraph: Troubleshoot Database Instance performance issues in Managed Databases for MySQL and PostgreSQL.
tags: disk-full databases
dates:
  validation: 2025-01-29
  posted: 2025-01-29
categories:
  - managed-databases
  - postgresql-and-mysql
---

## Problem

My Database Instance is not performing as expected.

## Possible causes

- High data loads
- High incoming traffic from specific apps or websites
- Huge or sub-optimized SQL queries

## Solution

You can carry out the following actions:

1. [Monitor usage metrics on Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit) to:

    - Check if the usage is nominal. The Database Instance can be impacted by high data loads or unexpected high traffic from an app or website, for example. You can [upgrade your Instance](/managed-databases-for-postgresql-and-mysql/how-to/upgrade-a-database) if necessary.
    - Check if there are slow queries in the Database Instance log. Huge or sub-optimized queries can slow down the instance significantly, or even fill up all the memory and trigger OOM kills.

> 2. Try connecting to the Database Instance in a [Private Network](/managed-databases-for-postgresql-and-mysql/how-to/connect-database-private-network). Public endpoint traffic for Managed Databases for MySQL and PostgreSQL goes through a load balancer, which adds significant latency. It also ensures better security, which makes it the recommended way to connect to a Managed Database.

Reviewer 3: I think its unclear whether the "it" in "It also ensures better security" if referring to public endpoint connection via load balancer, or over private network

> 2. Try connecting to the Database Instance in a [Private Network](/managed-databases-for-postgresql-and-mysql/how-to/connect-database-private-network). Public endpoint traffic for Managed Databases for MySQL and PostgreSQL goes through a load balancer, which adds significant latency. Using a Private Network also ensures better security, which makes it the recommended way to connect to a Managed Database.

Reviewer 4: ```suggestion
2. Try connecting to the Database Instance in a [Private Network](/managed-databases-for-postgresql-and-mysql/how-to/connect-database-private-network). Public endpoint traffic for Managed Databases for MySQL and PostgreSQL goes through a Load Balancer, which adds significant latency. Using a Private Network also ensures better security, which makes it the recommended way to connect to a Managed Database.
```
